core/src/main/java/org/jdbi/v3/BuiltInMapperFactory.java

```diff
@@ -129,6 +131,14 @@ private static URI getURI(ResultSet r, int i) throws SQLException {
         }
     }
 
+    private static UUID getUUID(ResultSet r, int i, StatementContext ctx) throws SQLException {
+        String s = r.getString(i);
+        if (s == null) {
+            return null;
+        }
+        return UUID.fromString(s);
```

Comment:
It's funny, when I looked closely at the Postgres JDBC driver, this is exactly what it's doing when you call getObject() on a UUID column.